troposphere/static/js/components/modals/instance/launch/steps/ImageSelectStep.react.js

```diff
@@ -65,7 +132,7 @@ export default React.createClass({
     },
 
     renderMoreImagesButton: function (images, totalNumberOfImages) {
-        if (images.models.length < totalNumberOfImages) {
+        if (images.length < totalNumberOfImages) {
```

Comment:
I think this madness of callbacks is a problem for this reason though. Probably better as components. This can be done in [ATMO-1242](https://pods.iplantcollaborative.org/jira/browse/ATMO-1242)

Steve had mentioned the goal of refactoring this into something other features could use. [ATMO-1242](https://pods.iplantcollaborative.org/jira/browse/ATMO-1242) being the ticket for that goal.